rasa_nlu/data_router.py

```diff
@@ -147,13 +149,16 @@ def extract(self, data):
     def parse(self, data):
         alias = data.get("model") or self.DEFAULT_MODEL_NAME
         if alias not in self.model_store:
-            raise InvalidModelError("No model found with alias '{}'".format(alias))
-        else:
-            model = self.model_store[alias]
-            response = model.parse(data['text'])
-            if self.responses:
-                self.responses.info(json.dumps(response, sort_keys=True))
-            return self.format_response(response)
+            try:
+                self.model_store[alias] = self.__interpreter_for_model(model_path=alias)
+            except:
+                raise InvalidModelError("No model found with alias '{}'".format(alias))
```

Comment:
Lines 88,89 also catch errors from `self.__interpreter_for_model` but log the actual error. That's what we should do here too